### DISCUSSION TITLE
Init new naked project in current directory?

### DISCUSSION THREAD

**Discussion by byrongibson** [facebook/docusaurus#10268] [Q&A]
Is there a way to initialize a new project in the current directory naked, instead of current directory/\<my-project\>/?

The use case is, I use Nix for development and have a project starter template directory that I copy and rename when starting any new project.  It looks like this:

```
| project-template
 \
  | scripts/
  | .envrc
  | .gitignore
  | default.nix
  | flake.nix
  | README.md
```

`direnv allow` reads flake.nix and installs node and yarn, among other things specified.  I would like to then be able run `npm init docusaurus@latest` (or similar) in that directory and have it result in the following:

```
| project-template
 \
  | .direnv/
  | .docusaurus/
  | .git/
  | .yarn/
  | blog/
  | docs/
  | scripts/
  | src/
  | static/
  | .envrc
  | .gitignore
  | .pnp.cjs*
  | .pnp.loader.mjs
  | babel.config.js
  | default.nix
  | docusaurus.config.ts
  | flake.lock
  | flake.nix
  | package-lock.json
  | package.json
  | README.md
  | sidebars.ts
  | tsconfig.json
  | yarn.lock
```

instead of what it does by default:

```
| project-template
 \
  | <my-docusaurus-project>/
  | scripts/
  | .envrc
  | .gitignore
  | default.nix
  | flake.nix
  | README.md
```

Possible?

**Comment by slorber** [comment]
I don't think there is a way currently, and the following fails: `npx create-docusaurus@latest . classic`


Would probably be easy to implement though, probably just changing this function:

```ts
  async function validateSiteName(siteName: string) {
    if (!siteName) {
      return 'A website name is required.';
    }
    const dest = path.resolve(rootDir, siteName);
    if (await fs.pathExists(dest)) {
      return logger.interpolate`Directory already exists at path=${dest}!`;
    }
    return true;
  }
```

But couldn't you just init in a subfolder and then run a command to move the files in parent folder?
